### PR TITLE
Separating slang-test in CI for emit-spirv-via-glsl

### PR DESCRIPTION
**Pull request by jkwak-work**
When CI test failes, we expect to see the error messages at the end of each section of CI testing result. But currently when "emit-spirv-via-glsl" fails, the error messages appear in the middle, which lead us to believe that there is no errors when you look only at the end of the result.

This commit separates out the "emit-spirv-via-glsl" case to make the error message more visiable.

### CODE REVIEW

**Comment by jkwak-work** [comment]
It appears that an option "-server-count 8" for slang-test was intentionally missing from Linux/macOS platforms.
I wonder if "-use-test-server" should also be removed from those platforms.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -113,20 +113,28 @@ jobs:
...
               failed=1
-          else
+          fi
+          exit $failed
+      - name: Test Slang spirv-via-glsl
+        run: |
+          export SLANG_RUN_SPIRV_VALIDATION=1
+          export SLANG_USE_SPV_SOURCE_LANGUAGE_UNKNOWN=1
+          failed=0
+          if [[ "${{matrix.full-gpu-tests}}" == "true" ]]; then
```
Comment (expipiplus1) [inline .github/workflows/ci.yml:130]: This conditional can be raised to be an `if` field alongside `name` and `run` now

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -113,20 +113,28 @@ jobs:
...
+              -expected-failure-list tests/expected-failure-github.txt ||
               failed=1
-          else
+          fi
+          exit $failed
+      - name: Test Slang spirv-via-glsl
+        run: |
+          export SLANG_RUN_SPIRV_VALIDATION=1
+          export SLANG_USE_SPV_SOURCE_LANGUAGE_UNKNOWN=1
+          failed=0
```
Comment (expipiplus1) [inline .github/workflows/ci.yml:129]: No need for the failed variable tracking now

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -113,20 +113,28 @@ jobs:
               -category ${{ matrix.test-category }} \
               -api all-cpu ||
               failed=1
```
Comment (expipiplus1) [inline .github/workflows/ci.yml:115]: No need for the failed variable tracking now

**Review by expipiplus1 — CHANGES_REQUESTED** [review]
lgtm aside from comments.

Although it might be neater to just synthesize the "via glsl" tests in the same way we synthesize cuda tests, that way we can dispense with having to run slang-test twice at all

File: .github/workflows/ci.yml (hunk 1 above)
Comment (expipiplus1) [reply .github/workflows/ci.yml:123]: I meant in the yaml having `if: matrix.full-gpu-tests == 'true'`

**Pull request merged**